sawmill-core/src/main/java/io/logz/sawmill/ProcessorFactoriesLoader.java

```diff
@@ -43,6 +46,11 @@ public void loadAnnotatedProcessors(ProcessorFactoryRegistry processorFactoryReg
                 processorFactoryRegistry.register(typeName, getFactory(processorProvider));
                 logger.debug("{} processor factory loaded successfully, took {}ms", typeName, stopwatch.elapsed(MILLISECONDS) - timeElapsed);
                 processorsLoaded++;
+            } catch(InvocationTargetException e) {
+                if (e.getCause() instanceof FactoryInstantiationException) {
+                    throw new FactoryInstantiationException(
+                            String.format("Could not instantiate %s processor factory", processor.getName()), e);
+                }
             } catch (Exception e) {
                 logger.error("failed to load processor {}", processor.getName(), e);
```

Comment:
@templatevk @asafm @barakm I am wondering - do we want to keep the exception swallowing in here? What benefit does it give?

I am thinking if maybe the proper logic is:
if dependencies are missing for a processor factory, `logger.warn` and continue
else, throw an exception

WDYT?

Reply:
i think that it depends on the processors you want to use, consider the following scenario:

1. I use 2 simple processors and I can't instantiate a pipeline cause of a buggy processor in the classpath
2. I wanna use geoip and I pass in the config, everything's good so far (silent log message) and I fail in runtime on processing the payload - i wish I really new this at the bootstrap so I could revert the deployment rather than have an error when actually processing my super important data

Reply:
So I suggest we preserve the current behavior - fail silently with a log message on all the processors and fail fast and explicitly on those who have Factory dependancies **provided incorrectly**

Reply:
I think you should fail. This is not a runtime library, it's something you place in your classpath, and a maybe add other dependencies that contribute processors. So if you can't instantiate a processor-factory that means, descriptors containing that processor won't be able to work. For me it sounds like a showstopper, that should make the library halt and not ignore.

Reply:
@templatevk the downside with the current behaviour is that it requires the processor author to recognize a special type of exception `FactoryInstantiationException` and throw it in order to fail the pipeline-factory

Reply:
That needs fixing. Any exception thrown, should fail pipeline-factory. It's a very easy fix.